src/pal/inc/pal.h

```diff
@@ -2411,13 +2411,13 @@ TlsFree(
 
 PALIMPORT
 void *
-PALAPI
-PAL_GetStackBase(VOID);
+PALAPI_NOEXPORT
+PAL_GetStackBase(VOID) DLLEXPORT;
```

Comment:
What makes these special that they need `DLLEXPORT` at the end?

Reply:
GCC was complaining `error: 'visibility' attribute ignored on non-class types`. In this order, attribute was being applied to the return type, instead of the function itself. The other way to fix this is to apply `PALAPI` before the return type.

In hindsight, just figured that except for `PAL_GetNativeExceptionHolderHead`, the rest of 15 cases spotted by GCC are those where we do not need `PALAPI`, therefore I have removed it in 730fc23, and changed the order of `PALAPI` (before the return type).

complete diff without whitespace changes: https://github.com/dotnet/coreclr/pull/22586/files?w=1